crates/matrix-sdk-crypto/src/olm/account.rs

```diff
@@ -424,6 +434,17 @@ impl Account {
         Self::new_helper(account, user_id, &device_id)
     }
 
+    /// Create a new random Olm Account for a dehydrated device
+    pub fn new_dehydrated(user_id: &UserId) -> Self {
+        let account = InnerAccount::new();
+        let device_id: OwnedDeviceId =
+            base64_encode(account.identity_keys().curve25519.as_bytes()).into();
```

Comment:
I meant to ask earlier, but on the MSC, I suggested using URL-safe unpadded Base64 instead of plain Base64, since the device ID gets put in path parameters.  If you don't have any objections to that, I may as well change this code to make this device ID URL-safe before we merge.

Reply:
We were pondering this when we did the iOS implementation and decided against it. The reason being that it would be confusing to have two different representations for a Curve25510 key. One being in the path and one being uploaded as the device keys. Both encodings would end up in the logs as well and confuse potential bug hunters.